core/frontend/src/AccuSnap.ts

```diff
@@ -37,9 +38,11 @@ export class TouchCursor implements CanvasDecoration {
   protected constructor(vp: ScreenViewport) {
     this._size = vp.pixelsFromInches(0.3);
     this._yOffset = this._size * 1.75;
+    this.viewport = vp;    
   }
 
   protected setPosition(vp: Viewport, worldLocation: Point3d): boolean {
+    this.viewport = vp;

```

Comment:
This should be moved right before vp.invalidateDecorations (i.e. after all the false early returns).

Reply:
OK, I pushed a fix.